### PR TITLE
feat(components): migrate List to TypeScript

### PR DESCRIPTION
**Pull request by connor-baer**
Addresses #516.

## Purpose

From #525:
> As explained in [this article](https://spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components/) and many others, enum props are preferable to boolean flags as they're mutually exclusive and easier to type.

## Approach and changes

- migrate List component to TypeScript
- replace `ordered` boolean prop with `variant` enum prop
- increase the default size to `mega` to match the Text component

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: src/components/List/List.spec.tsx [hunk 1]
```diff
@@ -15,74 +15,59 @@
...
+        <li>{size}</li>
       </List>
     );
-    expect(list).toMatchSnapshot();
+    expect(actual).toMatchSnapshot();
   });
 
-  it('should render nested unordered lists', () => {
-    const list = create(
+  it('should a render nested list', () => {
```
Comment (dmitri-suvorov-sumup) [inline src/components/List/List.spec.tsx:55]: ~should a render~ should render a

File: src/components/List/List.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2019, SumUp Ltd.
```
Comment (dmitri-suvorov-sumup) [inline src/components/List/List.tsx:2]: Can we update the copyright to 2020?

Comment (connor-baer) [reply src/components/List/List.tsx:2]: Since the initial code was written in 2019, it should remain as 2019.

**Comment by dmitri-suvorov-sumup** [comment]
Well done!

**Pull request merged**